### DISCUSSION TITLE
Timeline

### DISCUSSION THREAD

**Discussion by robertjmccarthy** [ministryofjustice/moj-frontend#716] [RELEASED: Components, pages and patterns]
The [timeline component](https://design-patterns.service.justice.gov.uk/components/timeline/) is live in the MoJ Design System.

Use this discussion to suggest changes and improvements to this component.

**Comment by Whersi** [comment]
## Ticket number

https://github.com/ministryofjustice/moj-frontend/issues/288

The details below is pulled from the ticket.

![image](https://user-images.githubusercontent.com/51957827/147080422-4b43796e-b302-48c2-840e-229e17cd07a2.png)

as seen in the screenshot, is it possible to close/end the timeline for the last item?  If you add to the timeline in reverse order, it does look weird with the oldest item having no end on the timeline. Or for the option to switch the end of the timeline (hanging line) to the beginning

**Comment by Whersi** [comment]
## Ticket number

https://github.com/ministryofjustice/moj-frontend/issues/44

The timeline component accepts a datetime object. For example:
```
datetime: {
  timestamp: "2019-06-14T14:01:00.000Z",
  type: "datetime"
}
```
Users have to include a `type` or `format` for the `timestamp`. Consider defaulting the type to `datetime` to that the user doesn't need to specify for each instance of an item.

**Comment by ClaireWaterworth** [comment]
I work on a caseworking service where there are known needs to include multiple nested events in timeline items (so the user doesn't have to scroll through many items to understand what has been updated on the service, and so that each update a user makes in one session is bundled into one event). I've done some Figma variations of this, which has tested well in our initial testing. We're soon going to explore this in more detail but I wondered if there are plans to explore the pattern with some more variations for each content block/timeline item. 
<img width="403" alt="Screenshot 2024-12-16 at 16 25 09" src="https://github.com/user-attachments/assets/b10369d1-aeb2-49ed-a509-13c763e4d25a" />

**Comment by Whersi** [comment]
@ClaireWaterworth The Timeline component is one of the next five components we plan to improve in the design system. As part of our process for improving or building a new component, we conduct a discovery phase. This involves gathering insights through surveys, 1:1 sessions, and information shared on platforms like Slack and GitHub discussions, such as this one. These efforts help us understand how the component is being used across MoJ, enabling us to define the scope of the design more effectively.

For the Timeline component, we will explore variations for each block or timeline item. If you could share any additional variations you’ve designed and keep us updated on your progress, it would be greatly appreciated.

**Reply by alicewilliams00** [reply]
Hey @Whersi, super interested to hear what you find. Will you be sharing findings here?

**Reply by Whersi** [reply]
> Hey @Whersi, super interested to hear what you find. Will you be sharing findings here?

We usually share the components we've worked on via GitHub, but we don’t always go into detail about our findings there.

If you'd like, drop us your email and we’ll invite you to our monthly Show & Tell sessions, where we do share more in-depth updates.

You’re also welcome to join our UK Government staff Slack channel: [#moj-design-system](https://ukgovernmentdigital.slack.com/archives/CJ6QDRDGC).

If you have any questions about the Design System, someone from the team will do their best to get back to you.

**Comment by fabiafowler** [comment]
Hello, do we have a concept of 'see more' for timelines that become extremely long?

**Reply by davehaigh** [reply]
Would be nice to explore this idea further. Especially as timelines are generally used on narrower columns e.g. 1/3 and paragraphs are likely to wrap alot more than when inside a normal 2/3 width column.

**Comment by AtinDESNEZ** [comment]
Is there a limit to, how many items would be appropriate to list (vertically) on the timeline component.

**Comment by helennickols** [comment]
When the code is next improved, it's proposed that we change the name from 'byline' to 'author' in the code. This is in-keeping with the documentation and plainer English.

**Comment by froddd** [comment]
Docs for the Nunjucks component show the `datetime` property as 'not required', but not including it results in 'Invalid date' being displayed. Is this a documentation issue, or a bug in implementation? Happy to open issue and PR if the latter.

**Reply by froddd** [reply]
Update: the wording 'Invalid date' comes from using our own date formatter, but the question attached is still relevant 🙂

**Reply by helennickols** [reply]
Thanks @froddd - this is a documentation issue. `datetime` property is required. I'll change this now.
